lib/src/main/kotlin/org/walletconnect/impls/MoshiPayloadEncryption.kt

```diff
@@ -0,0 +1,82 @@
+package org.walletconnect.impls
+
+import com.squareup.moshi.Moshi
+import org.bouncycastle.crypto.digests.SHA256Digest
+import org.bouncycastle.crypto.engines.AESEngine
+import org.bouncycastle.crypto.macs.HMac
+import org.bouncycastle.crypto.modes.CBCBlockCipher
+import org.bouncycastle.crypto.paddings.PKCS7Padding
+import org.bouncycastle.crypto.paddings.PaddedBufferedBlockCipher
+import org.bouncycastle.crypto.params.KeyParameter
+import org.bouncycastle.crypto.params.ParametersWithIV
+import org.komputing.khex.decode
+import org.komputing.khex.extensions.toNoPrefixHexString
+import org.walletconnect.Session
+import java.security.SecureRandom
+
+class MoshiPayloadEncryption(moshi: Moshi) : Session.PayloadEncryption {
+
+    private val payloadAdapter = moshi.adapter(MoshiPayloadAdapter.EncryptedPayload::class.java)
```

Comment:
nit: Let's name this `encryptedPayloadAdapter` since we are separating this `MoshiPayloadEncryption` class away from the `MoshiPayloadAdapter` class. The `payloadAdapter` name makes it seems like it is or should belong with a `Session.PayloadAdapter`.